src/checkqueue.h

```diff
@@ -96,7 +98,9 @@ class CCheckQueue
                         return fRet;
                     }
                     nIdle++;
+                    if (interrupted) return false;
                     cond.wait(lock); // wait
```

Comment:
Add the condition as a callback to wait.

e.g., cond.wait(lock, [&] { return interrupted || !queue.empty(); })